### PR TITLE
add CMake option TESTS

### PR DESCRIPTION
**Pull request by ranocha**
Cf. #28.

Shall we adapt the build flags before making the first release? Cf. #15.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -49,29 +51,51 @@ IF(USENVML)
   add_definitions(-DUSENVML)
 ENDIF()
 
-# Because Makefiles are shit to read
+set(CMAKE_CONFIGURATION_TYPES "Release" CACHE STRING "" FORCE)
```
Comment (ranocha) [inline CMakeLists.txt:54]: I think that part broke the Appveyor build on Windows: The target `ALL_BUILD` seems to include a debug build for which no info can be found.

**Comment by ranocha** [comment]
I don't understand why the linux build on Appveyor fails. There seem to be problems with the MSR registers that haven't been there before (https://ci.appveyor.com/project/ranocha/toolkiticl/builds/23065483/job/30jget4xg01si9pg#L2212).

File: CMakeLists.txt (hunk 1 above)
Comment (philipheinisch) [reply CMakeLists.txt:54]: On my system ALL_BUILD works and just creates a `Release` build. Maybe some strange CMake version problem?

Where does the error `read_msr():pread: Input/output error` come from?

**Comment by ranocha** [comment]
For the failed build on Linux, `clinfo` reports a slightly different CPU: `Intel(R) Xeon(R) CPU @ 2.30GHz` (failed) instead of `Intel(R) Xeon(R) CPU E5-2697 v3 @ 2.60GHz` (pass).

File: CMakeLists.txt (hunk 1 above)
Comment (ranocha) [reply CMakeLists.txt:54]: I've fixed the problem on Windows. We had to add the argument `--config Release` to `cmake`.

**Comment by ranocha** [comment]
Maybe reading MSR is simply not reliable on virtual platforms. I'll disable that test.

**Comment by philipheinisch** [comment]
> For the failed build on Linux, `clinfo` reports a slightly different CPU: `Intel(R) Xeon(R) CPU @ 2.30GHz` (failed) instead of `Intel(R) Xeon(R) CPU E5-2697 v3 @ 2.60GHz` (pass).

In virtual environments, this difference is in most cases caused by simulating a generic architecture (hence no CPU type) instead of directly using the host CPU (the exact type is displayed). Virtualizing generic CPUs gives a lot more flexibility from an administrative standpoint and is preferred on most shared resource systems, as VMs can easily be moved between different host systems and the workload can be shares between different CPUs. The drawback is, that no CPU specific functions (like MSR) are available.

**Comment by philipheinisch** [comment]
Can we merge this now or do we want to add more as part of this PR?

**Comment by ranocha** [comment]
I think you can merge this now.

**Pull request merged**